libcudacxx/include/cuda/std/detail/libcxx/include/complex

```diff
@@ -292,6 +292,21 @@ template<class T> complex<T> tanh (const complex<T>&);
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
+template<class _Tp>
```

Comment:
Could we introduce an `__is_half_floating_point` trait that we use whenever we want to enable some specializations?

Reply:
I haven't seen the need. The case of specializing some of the operations for vector-enabled types should use a separate trait, because it's not impossible that eventually we get wider vector ops that we'll want to hook into in there.